mass_mailing_switzerland/models/mass_mailing_campaign.py

```diff
@@ -26,9 +26,20 @@ class MassMailingCampaign(models.Model):
         'correspondence', related='campaign_id.correspondence_ids'
     )
     invoice_line_ids = fields.One2many(
-        'account.invoice.line', related='campaign_id.invoice_line_ids'
+        'account.invoice.line', compute='_compute_campaign_invoices'
     )
 
+    @api.multi
+    def _compute_campaign_invoices(self):
+        for campaign in self:
+            invoices = []
```

Comment:
Instead of your solution I propose this which is shorter:
```suggestion
            mass_medium = self.env.ref('contract_compassion.utm_medium_mass_mailing')
            campaign.invoice_line_ids = campaign.campaign_id.invoice_line_ids\
                .filtered(lambda invl: invl.medium_id == mass_medium)
```